libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -485,7 +485,10 @@ export class FunctionUtils {
      * @param verify Function to check each arg for validity.
      * @returns Delegate for evaluating an expression.
      */
-    public static applyWithError(func: (arg0: any[]) => any, verify?: VerifyExpression): EvaluateExpressionDelegate {
+    public static applyWithError(
+        func: (arg0: any[]) => ValueWithError,
```

Comment:
Do these need to be `any`? Can they just be `unknown`?

Reply:
I don't think it can be unkown here. The implicit any here means it argument can be any type and the return value also can be any type. Changing any to unkown here will cause unexpect error from tsc.

Reply:
From an API perspective, `any` and `unknown` are quite similar. If a function has a parameter that is of type `unknown`, a developer can pass any type into the function. The benefit of `unknown` is that, within the function, the Typescript compiler will not allow the author to assume anything about the type of the parameter.

That means, in order to call a string function on an `unknown` value, you must first call `if (typeof val === 'string')`. In general it provides more type safety and less chance for runtime exceptions related to improper types.

Here's a quick Typescript playground sample to illustrate: https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAQzATwGJgBQDdkA2AXCugJQkDOUATjGAOaIA+iYIBBiA3gLABQiRDQCmUEDST4CAOihwAMnADuImgGFklEdjIBuAQF8BA0JFgJE4ANZgVYAGqE8hEjbvKwFRNTqMWbBxcfILCYhJShHKKKmqa2roGoQD0yWHikohQaAAOInDAiNKIALxliADkvvQMFYgA-EVR8kqqGlo6ZIgk7JxJhkA

Reply:
got it. I changed the ApplyWithOptions and ApplyWithOptionsAndError. The other methods will have a seperate PR to fix that.